Sources/PostgresNIO/New/Data/Bool+PSQLCodable.swift

```diff
@@ -3,18 +3,42 @@ extension Bool: PSQLCodable {
         .bool
     }
     
-    static func decode(from buffer: inout ByteBuffer, type: PSQLDataType, context: PSQLDecodingContext) throws -> Bool {
-        guard type == .bool, buffer.readableBytes == 1 else {
+    var psqlFormat: PSQLFormat {
+        .binary
+    }
+    
+    static func decode(from buffer: inout ByteBuffer, type: PSQLDataType, format: PSQLFormat, context: PSQLDecodingContext) throws -> Bool {
+        guard type == .bool else {
             throw PSQLCastingError.failure(targetType: Self.self, type: type, postgresData: buffer, context: context)
         }
         
-        switch buffer.readInteger(as: UInt8.self) {
-        case .some(0):
-            return false
-        case .some(1):
-            return true
-        default:
-            throw PSQLCastingError.failure(targetType: Self.self, type: type, postgresData: buffer, context: context)
+        switch format {
+        case .binary:
+            guard buffer.readableBytes == 1 else {
```

Comment:
(This question applies to a handful of other places in the code too.)

Reply:
Because more than one byte would mean that we got data that we don't expect. In the case of a bool we expect exactly one byte. If that's not true anymore or cast method needs adjustment.